### PR TITLE
document iterator structs

### LINKED ISSUE #9: Document structs
- [ ] `Iter`
- [ ] `IterMut`
- [ ] `Keys`
- [ ] `Values`

### PR DESCRIPTION
**Pull request by apasel422**
closes #9

### CODE REVIEW

**Comment by FlashCat** [comment]
r? @bluss

(I've picked a reviewer for you, use r? to override)

**Comment by bluss** [comment]
Hey! With FlashCat you just merge yourself, right?

File: src/lib.rs [hunk 1]
```diff
@@ -28,6 +28,7 @@
 //! assert_eq!(items, [(2, 20), (1, 10), (3, 30)]);
 //! ```
 
+#![forbid(missing_docs)]
```
Comment (bluss) [inline src/lib.rs]: didn't want to hold you up, but there's not much difference between `deny` and `forbid` here is there?

Comment (apasel422) [inline src/lib.rs]: I think I accidentally typed `forbid`, but the difference is as follows:
> [forbid(C) is the same as deny(C), but also forbids changing the lint level afterwards](https://doc.rust-lang.org/reference.html#lint-check-attributes)

We can change to `deny` later, if need be.

**Pull request merged**